stock_release_channel_process_end_time_shipment_lead_time/tests/test_release_end_date.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from freezegun import freeze_time
+
+from odoo import fields
+
+from odoo.addons.queue_job.job import Job
+from odoo.addons.stock_release_channel.tests.common import ChannelReleaseCase
+
+
+class ReleaseChannelEndDateCase(ChannelReleaseCase):
+    @freeze_time("2023-01-27 10:00:00")
+    def test_channel_counter_release_ready(self):
+        self.env["ir.config_parameter"].sudo().set_param(
+            "stock_release_channel_process_end_time.stock_release_use_channel_end_date",
+            True,
+        )
+        # Remove existing jobs as some already exists to assign pickings to channel
+        jobs_before = self.env["queue.job"].search([])
+        jobs_before.unlink()
+        # Set the end time
+        self.channel.process_end_time = 23.0
+        # Set picking scheduled date
+        pickings = self.picking | self.picking2 | self.picking3
+        pickings.write({"scheduled_date": fields.Datetime.now()})
+
+        # Asleep the release channel to void the process end date
+        self.channel.action_sleep()
+        self.channel.invalidate_recordset()
+        self.channel.action_wake_up()
+        # Execute the picking channel assignations
+        jobs_after = self.env["queue.job"].search([])
+        for job in jobs_after:
+            job = Job.load(job.env, job.uuid)
+            job.perform()
```

Comment:
You can use the `test_queue_job_no_delay` context key to run the jobs in the same transaction:
```suggestion
        # Execute the picking channel assignations
        self.channel.with_context(test_queue_job_no_delay=True).action_wake_up()
```